src/wp-includes/class-wp-user-query.php

```diff
@@ -840,6 +840,9 @@ public function query() {
 				$result->id = $result->ID;
 			}
 		} elseif ( 'all_with_meta' === $qv['fields'] || 'all' === $qv['fields'] ) {
+			if ( ! function_exists( 'cache_users' ) ) {
+				require_once ABSPATH . WPINC . '/pluggable.php';
+			}
 			cache_users( $this->results );
```

Comment:
Maybe something like this?

```suggestion
			if ( function_exists( 'cache_users' ) ) {
				cache_users( $this->results );
			} else {
				_doing_it_wrong(
					'WP_User_Query',
					sprintf(
						/* translators: plugins_loaded */
						__( 'User queries should not be run before the %s hook.' ),
						'<code>plugins_loaded</code>'
					);
					'6.1.1'
				);
			}
			
```

Reply:
+1 to this fix.